crml/transaction-payment/src/constants.rs

```diff
@@ -0,0 +1,53 @@
+// Copyright 2018-2020 Parity Technologies (UK) Ltd. and Centrality Investments Ltd.
+// This file is part of Substrate.
+
+// Substrate is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+
+// Substrate is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+
+// You should have received a copy of the GNU General Public License
+// along with Substrate.  If not, see <http://www.gnu.org/licenses/>.
+
+//! # Transaction Payment Module
+//!
+//! Transaction Payment Customized Error Code Constants
+
+pub(crate) mod error_code {
+	pub const CORE_TO_ASSET_PRICE_NOT_ABOVE_ZERO: u8 = 190;
+	pub const CORE_TO_ASSET_PRICE_ABOVE_MAX_LIMIT: u8 = 191;
+	pub const INSUFFICIENT_BUYER_CORE_ASSET_BALANCE: u8 = 192;
+
+	pub const ASSET_TO_CORE_PRICE_NOT_ABOVE_ZERO: u8 = 193;
+	pub const ASSET_TO_CORE_PRICE_ABOVE_MAX_LIMIT: u8 = 194;
+	pub const INSUFFICIENT_BUYER_TRADE_ASSET_BALANCE: u8 = 195;
+
+	pub const ASSET_SALE_VALUE_NOT_ABOVE_ZERO: u8 = 196;
+	pub const SALE_VALUE_BELOW_REQUIRED_MINIMUM: u8 = 197;
+	pub const INSUFFICIENT_SELLER_CORE_ASSET_BALANCE: u8 = 198;
+	pub const BUY_AMOUNT_NOT_POSITIVE: u8 = 199;
+	pub const INVALID_ASSET_ID: u8 = 200;
+	pub const UNKNOW_BUY_FEE_ASSET: u8 = 201;
+
+	pub fn buy_fee_asset_error_to_code(error: u8) -> u8 {
+		match error {
+			4 => INSUFFICIENT_BUYER_TRADE_ASSET_BALANCE,
```

Comment:
Is it possible to import CENNZ-X error type here and match over that instead of raw number?

Reply:
This is the struct for DispatchError:
```
pub enum DispatchError {
	/// Some error occurred.
	Other(#[codec(skip)] &'static str),
	/// Failed to lookup some data.
	CannotLookup,
	/// A bad origin.
	BadOrigin,
	/// A custom error in a module
	Module {
		/// Module index, matching the metadata module index
		index: u8,
		/// Module specific error value
		error: u8,
		/// Optional error message.
		#[codec(skip)]
		message: Option<&'static str>,
	},
}
```

So we can use `message` instead of `error` number for matching.
It will be like this:

```
pub fn buy_fee_asset_error_to_code(message: &'static str) -> u8 {
		match message {
			“InsufficientBuyerTradeAssetBalance” => INSUFFICIENT_BUYER_TRADE_ASSET_BALANCE,
```